packages/scheduler/lib/scheduler.ts

```diff
@@ -383,13 +383,17 @@ export class Scheduler {
      */
     public async setScheduleFrequency({ scheduleName, frequencyMs }: { scheduleName: string; frequencyMs: number }): Promise<Result<Schedule>> {
         return this.dbClient.db.transaction(async (trx) => {
-            const schedule = await schedules.search(trx, { names: [scheduleName], limit: 1, forUpdate: true });
```

Comment:
we are doing a single update below so no need to lock the schedule. One could argue that we don't need the transaction anymore